bin/git-forgit

```diff
@@ -104,6 +104,29 @@ _forgit_parse_array() {
     ${old_IFS+"false"} && unset IFS || IFS="$old_IFS"
 }
 
+# parse the input arguments and print only those after the "--"
+# separator as a single line of qouted arguments to stdout
+_forgit_quote_files() {
+    local files add
+    files=()
+    add=false
+    while (( "$#" )); do
+        case "$1" in
+            --)
+            add=true
+            shift
+            ;;
+            *)
+            if [ $add == true ]; then
+                files+=("'$1'")
```

Comment:
Forgive my ignorance on this more advanced shell syntax.

It seems like 114 our case statement is "I found a --, set add to true, and shift (meaning that positional arguments are shifted one to the left" then we add it to the array.


What happens with a list of files, aka:

`glo -- foo.txt bar.txt` Wouldn't the second one be skipped? Do we expect people to use `glo -- foo.txt -- bar.txt`?

(Completely possible I'm missing the main understanding here, just trying to understand what this means for the user)

Reply:
No worries, bash syntax can be a bit confusing sometimes.

> It seems like 114 our case statement is "I found a --, set add to true, and shift (meaning that positional arguments are shifted one to the left" then we add it to the array.

Yes, that's correct. 

> What happens with a list of files, aka:
> glo -- foo.txt bar.txt Wouldn't the second one be skipped? 

No, let me walk you through this example:

The while loop runs as long as there are any there are any positional arguments present. In your example we have 3 of them ("--", "foo.txt" and "bar.txt").

1. Iteration:
The first positional argument (`$1`) has a value of  "--". This matches the first case: `$add` is set `true`, we shift and terminate the first iteration. Keep in mind that when using `;;`, the case statement does not "fall through",meaning the iteration ends without evaluating the other cases (this would only be the case when `;;&` is used as a terminator).
2. Iteration:
`$1` has a value of "foo.txt". This does not match the first case. It does match the second case (which simply matches everything). Since `$add` is `true`, "'foo.txt'" is added to the `$files` array and we shift again.
3. Iteration
`$1` is now "bar.txt" and the same thing happens as in the second one. "'bar'" is added to the array and we shift, causing `$#` to become 0 and terminate the while loop.

This implementation also ensures only arguments that come after -- are treated as files and will be added to the array, since everything before -- will simply be skipped over, since `$add` is still `false`.

Does that answer your question?

Reply:
Thanks for the extremely in depth walkthrough! Appreciate it. Makes sense to me!